vm/src/hint_processor/hint_processor_definition.rs

```diff
@@ -52,8 +49,27 @@ pub trait HintProcessorLogic {
             ids_data: get_ids_data(reference_ids, references)?,
         }))
     }
+
+    //Executes the hint which's data is provided by a dynamic structure previously created by compile_hint
+    // Can also return a map of hints to be loaded after the current hint is executed
+    fn execute_hint_extensive(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        //Data structure that can be downcasted to the structure generated by compile_hint
+        hint_data: &Box<dyn Any>,
+        //Constant values extracted from the program specification.
+        constants: &HashMap<String, Felt252>,
+    ) -> Result<Option<HintExtension>, HintError> {
+        self.execute_hint(vm, exec_scopes, hint_data, constants)?;
+        Ok(None)
+    }
 }
 
+// A map of hints that can be used to extend the current map of hints for the vm run
+// The map matches the pc at which the hints should be executed to a vec of compiled hints (Outputed by HintProcessor::CompileHint)
+pub type HintExtension = Box<HashMap<Relocatable, Vec<Box<dyn Any>>>>;
```

Comment:
I think we can avoid the outer `Box`. The `HashMap` already stores its contents on the heap and will be moved.
```suggestion
pub type HintExtension = HashMap<Relocatable, Vec<Box<dyn Any>>>;
```